hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/pipeline/PipelineManagerV2Impl.java

```diff
@@ -136,6 +139,9 @@ public static PipelineManagerV2Impl newPipelineManager(
   @Override
   public Pipeline createPipeline(ReplicationType type,
                                  ReplicationFactor factor) throws IOException {
+    if (!scmhaManager.isLeader()) {
```

Comment:
NIT: Can we add a helper function like checkLeader() to dedup the code?

Reply:
Updated.